src/framework/components/anim/controller.js

```diff
@@ -0,0 +1,638 @@
+                    case ANIM_TRANSITION_PREDICATE_GREATER_THAN:
+                        conditionsMet = conditionsMet && parameter.value > condition.value;
+                        break;
+                    case ANIM_TRANSITION_PREDICATE_LESS_THAN:
+                        conditionsMet = conditionsMet && parameter.value < condition.value;
+                        break;
+                    case ANIM_TRANSITION_PREDICATE_GREATER_THAN_EQUAL_TO:
+                        conditionsMet = conditionsMet && parameter.value >= condition.value;
+                        break;
+                    case ANIM_TRANSITION_PREDICATE_LESS_THAN_EQUAL_TO:
+                        conditionsMet = conditionsMet && parameter.value <= condition.value;
+                        break;
+                    case ANIM_TRANSITION_PREDICATE_EQUAL_TO:
+                        conditionsMet = conditionsMet && parameter.value === condition.value;
+                        break;
+                    case ANIM_TRANSITION_PREDICATE_NOT_EQUAL_TO:
+                        conditionsMet = conditionsMet && parameter.value !== condition.value;
+                        break;
+                }
+                if (!conditionsMet)
+                    return conditionsMet;
+            }
+            return conditionsMet;
+        }
+    });
+
+    var AnimController = function (animEvaluator, states, transitions, parameters, activate) {
+        this._animEvaluator = animEvaluator;
+        this._states = {};
+        var i;
+        for (i = 0; i < states.length; i++) {
+            this._states[states[i].name] = new AnimState(
+                this,
+                states[i].name,
+                states[i].speed
+            );
+        }
+        this._transitions = transitions.map(function (transition) {
+            return new AnimTransition(
+                this,
+                transition.from,
+                transition.to,
+                transition.time,
+                transition.priority,
+                transition.conditions,
+                transition.exitTime,
+                transition.transitionOffset,
+                transition.interruptionSource
+            );
+        }.bind(this));
+        this._findTransitionsFromStateCache = {};
+        this._findTransitionsBetweenStatesCache = {};
+        this._parameters = parameters;
+        this._initialParameters = Object.assign({}, parameters);
+        this._previousStateName = null;
+        this._activeStateName = ANIM_STATE_START;
+        this._playing = false;
+        this._activate = activate;
+
+        this._currTransitionTime = 1.0;
+        this._totalTransitionTime = 1.0;
+        this._isTransitioning = false;
+        this._transitionInterruptionSource = ANIM_INTERRUPTION_SOURCE_NONE;
+        this._transitionPreviousStates = [];
+
+        this._timeInState = 0;
+        this._timeInStateBefore = 0;
+    };
+
+    Object.defineProperty(AnimController.prototype, 'animEvaluator', {
```

Comment:
Just in case you haven't seen already... it is possible to define multiple properties together in a single block. I quite like this because it cuts down on some of the boilerplate code and also groups the properties together visually.

(This is just a passing comment btw, not a formal request-for-change :)

For an example of it's structure see https://github.com/playcanvas/engine/blob/master/src/anim/anim.js#L48

Reply:
Less boilerplate is always good! I've updated it to use this.